Source/Core/Core.swift

```diff
@@ -481,11 +481,36 @@ public class FormViewController : UIViewController, FormViewControllerProtocol {
     
     public override func viewWillAppear(animated: Bool) {
         super.viewWillAppear(animated)
-        if let selectedIndexPath = tableView?.indexPathForSelectedRow {
-            tableView?.reloadRowsAtIndexPaths([selectedIndexPath], withRowAnimation: .None)
-            tableView?.selectRowAtIndexPath(selectedIndexPath, animated: false, scrollPosition: .None)
-            tableView?.deselectRowAtIndexPath(selectedIndexPath, animated: true)
+
+        let selectedIndexPaths = tableView?.indexPathsForSelectedRows ?? []
+        tableView?.reloadRowsAtIndexPaths(selectedIndexPaths, withRowAnimation: .None)
+        selectedIndexPaths.forEach {
+            tableView?.selectRowAtIndexPath($0, animated: false, scrollPosition: .None)
+        }
+
+        let deselectionAnimation = { (context: UIViewControllerTransitionCoordinatorContext) in
```

Comment:
Got it. I'm not completely sure how swift manage these references and which one is released first. I've already added the `[weak self]`, the result should be the same. Note that i didn't add it to foreach since it explicitly declare `@noescape`.